File: 0299 Bulls and Cows.java
class Solution {
    public String getHint(String secret, String guess) {
        int[] s = new int[10];
        int[] g = new int[10];
        int l = secret.length();
        int cntA = 0;
        int cntB = 0;
        for (int i = 0; i < l; ++i) {
            int n1 = secret.charAt(i) - '0', n2 = guess.charAt(i) - '0';
            if (n1 == n2) {
                ++cntA;
            } else {
                ++s[n1];
                ++g[n2];
            }
        }
        for (int i = 0; i < 10; ++i) {
            if (s[i] <= g[i]) {
                cntB += s[i];
            } else {
                cntB += g[i];
            }                
        }
        return cntA + "A" + cntB + "B";
    }
}

File: 0299 Bulls and Cows.py
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        s, g = [0]*10, [0]*10
        l = len(secret)
        a, b = 0, 0
        
        for i in range(l):
            n1, n2 = int(secret[i]), int(guess[i])
            if n1 == n2:
                a += 1
            else:
                s[n1] += 1
                g[n2] += 1
                
        for i in range(10):
            b += min(s[i], g[i])
        
        return str(a) + "A" + str(b) + "B"
''' Solution from leetcode
from collections import Counter
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        s = Counter(secret)
        g = Counter(guess)
        secretList = list(secret)
        guessList = list(guess)
        bull, cow = 0,0
        for i, val  in enumerate(guessList):
            if secretList[i] == val: bull+=1
        cow = sum((s & g).values()) - bull
        return str(bull)+'A'+str(cow)+'B'
'''